app/controllers/api/service_requests_controller.rb

```diff
@@ -44,5 +44,17 @@ def service_requests_search_conditions
       return {} if @auth_user_obj.admin?
       {:requester => @auth_user_obj}
     end
+
+    def add_approver_resource(type, id, data)
+      user_id = data['user_id'] || parse_id(data['user'], :users)
+      raise 'Must provide a user ID' unless user_id
```

Comment:
I wonder if the message should  reflect either, maybe "Must specify a valid user_id or user", also need a test added for this exception.  Thanks.